Dapper/SqlMapper.Async.cs

```diff
@@ -1217,5 +1218,79 @@ private static async Task<T> ExecuteScalarImplAsync<T>(IDbConnection cnn, Comman
+        /// <param name="commandType">The type of command to execute.</param>
+        /// <returns>
+        /// A sequence of data of <typeparamref name="T"/>; if a basic type (int, string, etc) is queried then the data from the first column is assumed, otherwise an instance is
+        /// created per row, and a direct column-name===member-name mapping is assumed (case insensitive).
+        /// </returns>
+        public static IAsyncEnumerable<T> QueryUnbufferedAsync<T>(this DbConnection cnn, string sql, object param = null, DbTransaction transaction = null, int? commandTimeout = null, CommandType? commandType = null)
+        {
+            // note: in many cases of adding a new async method I might add a CancellationToken - however, cancellation is expressed via WithCancellation on iterators
+            return QueryUnbufferedAsync<T>(cnn, typeof(T), new CommandDefinition(sql, param, transaction, commandTimeout, commandType, CommandFlags.None, default));
+        }
+
+        private static IAsyncEnumerable<T> QueryUnbufferedAsync<T>(this IDbConnection cnn, Type effectiveType, CommandDefinition command)
+        {
+            return Impl(cnn, effectiveType, command, command.CancellationToken); // proxy to allow CT expression
+
+            static async IAsyncEnumerable<T> Impl(IDbConnection cnn, Type effectiveType, CommandDefinition command,
+                [EnumeratorCancellation] CancellationToken cancel)
+            {
+                object param = command.Parameters;
+                var identity = new Identity(command.CommandText, command.CommandType, cnn, effectiveType, param?.GetType());
+                var info = GetCacheInfo(identity, param, command.AddToCache);
+                bool wasClosed = cnn.State == ConnectionState.Closed;
+                using var cmd = command.TrySetupAsyncCommand(cnn, info.ParamReader);
+                DbDataReader reader = null;
+                try
+                {
+                    if (wasClosed) await cnn.TryOpenAsync(cancel).ConfigureAwait(false);
+                    reader = await ExecuteReaderWithFlagsFallbackAsync(cmd, wasClosed, CommandBehavior.SequentialAccess | CommandBehavior.SingleResult, cancel).ConfigureAwait(false);
+
+                    var tuple = info.Deserializer;
+                    int hash = GetColumnHash(reader);
+                    if (tuple.Func == null || tuple.Hash != hash)
+                    {
+                        if (reader.FieldCount == 0)
+                        {
+                            yield break;
+                        }
+                        tuple = info.Deserializer = new DeserializerState(hash, GetDeserializer(effectiveType, reader, 0, -1, false));
+                        if (command.AddToCache) SetQueryCache(identity, info);
+                    }
+
+                    var func = tuple.Func;
+
+                    var convertToType = Nullable.GetUnderlyingType(effectiveType) ?? effectiveType;

```

Comment:
This `convertToType` variable is not used. I guess it was meant to be used 4 lines below?

```cs
yield return GetValue<T>(reader, convertToType, val);
```

Reply:
You're almost certainly correct; thanks, good eyes! I'll try to look at this tomorrow (I guess I need some tests that do `QueryUnbuffered<int>`, `QueryUnbuffered<int?>`, `QueryUnbuffered<string>`, `QueryUnbuffered<SomeValueType>` and `QueryUnbuffered<SomeClass>`

Reply:
Or maybe it should just be deleted because apparently the `convertToType` is already computed inside `GetValue<T>(…)`.

Reply:
added for tracking: https://github.com/DapperLib/Dapper/issues/1920

Reply:
> good eyes

Actually, it's JetBrains Rider dimming unused variables. 😉

![image](https://github.com/DapperLib/Dapper/assets/51363/e32ae94c-d2cd-4815-b1e4-080de9a9c6da)